helix-core/src/test/java/org/apache/helix/messaging/handling/TestHelixTaskExecutor.java

```diff
@@ -232,11 +248,29 @@ public HelixTaskResult handleMessage() throws InterruptedException {
       @Override
       public void onError(Exception e, ErrorCode code, ErrorType type) {
       }
+
+      @Override
+      public void precheckForStaleMessage() throws Exception {
+        if (_testIsMessageStaled) {
+          super.precheckForStaleMessage();
+        }
+      }
     }
 
     @Override
     public MessageHandler createHandler(Message message, NotificationContext context) {
-      return new TestStateTransitionMessageHandler(message, context);
+      if (message.getResourceName()!="testStaledMessageResource") {
```

Comment:
`equals()` should be used to check the string names are the same.
`!=` checks if they are the **same** object: same memory address, etc..

Reply:
TFTR. Updated.